sqlmesh/core/loader.py

```diff
@@ -404,32 +411,64 @@ def _load() -> Model:
                     except SqlglotError as ex:
                         raise ConfigError(f"Failed to parse a model definition at '{path}': {ex}.")
 
-                return load_sql_based_model(
-                    expressions,
-                    defaults=self.config.model_defaults.dict(),
-                    macros=macros,
-                    jinja_macros=jinja_macros,
-                    audit_definitions=audits,
-                    default_audits=self.config.model_defaults.audits,
-                    path=Path(path).absolute(),
-                    module_path=self.config_path,
-                    dialect=self.config.model_defaults.dialect,
-                    time_column_format=self.config.time_column_format,
-                    physical_schema_mapping=self.config.physical_schema_mapping,
-                    project=self.config.project,
-                    default_catalog=self.context.default_catalog,
-                    variables=self._variables,
-                    infer_names=self.config.model_naming.infer_names,
-                    signal_definitions=signals,
-                )
+                model_meta = seq_get(expressions, 0)
+                blueprints = isinstance(model_meta, ModelMeta) and model_meta.meta.get("blueprints")
+
+                loaded_models = []
+                for blueprint in blueprints or [None]:
```

Comment:
This again leaks the details of model handling to the loader. I don't think the loader should worry about this. `load_sql_based_model` should encapsulate this logic instead.

Reply:
Implemented a `load_sql_based_models` helper and refactored the previous approach by pushing the logic to the `model/definition.py` module. Please take a look and let me know if you had something else in mind.